dai_cadcad/flip.py

```diff
@@ -0,0 +1,108 @@
+""" Flip Module
+    Class-based representation of the Flip smart contract
+    (contains only what is necessary for the simulation)
+"""
+
+
+class Flip:
+    ADDRESS = "flip"
+
+    vat = None
+    beg = None
+    ttl = None
+    tau = None
+    kicks = None
+    bids = {}
+
+    def __init__(self, vat, beg, ttl, tau, kicks, bids):
+        """"""
+
+        self.vat = vat
+
+        self.beg = beg
+        self.ttl = ttl
+        self.tau = tau
+        self.kicks = kicks
+        self.bids = bids
+
+    def flipper_kick(flipper, vat, user_id, gal, tab, lot, bid, end):
```

Comment:
For this and the rest of the functions, we should:
1. Remove the `flipper_` prefix
2. Remove `flipper`, `vat`, and any other function parameters no longer necessary in the OOP setup
  - Make the appropriate changes, like using `self.vat` instead of `vat` and `self` instead of `flipper`
  - Oh also add `self` as the first parameter everywhere
3. Rename `user_id` to `usr` to match the Maker code